polygeist/tools/cgeist/Lib/CodeGenTypes.cc

```diff
@@ -1474,7 +1474,7 @@ mlir::Type CodeGenTypes::getMLIRType(clang::QualType QT, bool *ImplicitRef,
     bool SubRef = false;
     auto ET = getMLIRType(AT->getElementType(), &SubRef, AllowMerge);
     int64_t Size = AT->getNumElements();
-    if (isa<clang::ExtVectorType>(T))
+    if (isa<clang::VectorType>(T) || isa<clang::ExtVectorType>(T))
```

Comment:
If we are in this region under `if (const auto *AT = dyn_cast<clang::VectorType>(T))`, `isa<clang::VectorType>(T)` should always be true, is it safe to just change this to `return mlir::VectorType::get(Size, ET);` directly, and remove the code after in this region.

Reply:
done